rust/catalyst-types/src/uuid/uuid_v4.rs

```diff
@@ -41,23 +43,18 @@ impl Display for UuidV4 {
     }
 }
 
-impl TryFrom<&coset::cbor::Value> for UuidV4 {
-    type Error = super::CborUuidError;
-
-    fn try_from(cbor_value: &coset::cbor::Value) -> Result<Self, Self::Error> {
-        match decode_cbor_uuid(cbor_value) {
-            Ok(uuid) => {
-                if uuid.get_version_num() == Self::UUID_VERSION_NUMBER {
-                    Ok(Self { uuid })
-                } else {
-                    Err(super::CborUuidError::InvalidVersion {
-                        uuid,
-                        expected_version: Self::UUID_VERSION_NUMBER,
-                    })
-                }
-            },
-            Err(e) => Err(e),
-        }
+impl Decode<'_, ()> for UuidV4 {
+    fn decode(d: &mut Decoder<'_>, ctx: &mut ()) -> Result<Self, minicbor::decode::Error> {
+        let uuid = decode_cbor_uuid(d, ctx)?;
```

Comment:
Should take the same context as suggested above, so decoding can either be forced to have a tag or not, or be optional.